content/docs/command-reference/remove.md

```diff
@@ -9,27 +9,30 @@ optionally delete them).
 usage: dvc remove [-h] [-q | -v] [--outs] targets [targets ...]
 
 positional arguments:
-  targets        stages (found in dvc.yaml) or .dvc files to remove.
+  targets        Tracked files/directories, stage names (found in
+                 dvc.yaml), or .dvc files to remove.
 ```
 
 ## Description
 
-Safely removes `.dvc` files or stages from `dvc.yaml`. This includes deleting
-the corresponding `.gitignore` entries (based on the `outs` fields removed).
+Safely removes tracked files/directories, stage names (found in `dvc.yaml`), or
+`.dvc` files. This includes deleting the corresponding `.gitignore` entries.
 
 > `dvc remove` doesn't remove files from the DVC <abbr>cache</abbr> or
 > [remote storage](/doc/command-reference/remote). Use `dvc gc` for that.
 
-It takes one or more stage names (see `-n` option of `dvc run`) or `.dvc` file
-names as `targets`.
+It takes one or more stage names (see `-n` option of `dvc run`), `.dvc` file
+names or tracked files/directories as `targets`.
 
 If there are no stages left in `dvc.yaml` after the removal, then both
 `dvc.yaml` and `dvc.lock` are deleted. `.gitignore` is also deleted if there are
 no more entries left in it.
 
-Note that the actual <abbr>output</abbr> files or directories of the stage
-(`outs` field) are not removed by this command, unless the `--outs` option is
-used.
+Note that, when using stage name as target, the actual <abbr>output</abbr> files
+or directories of the stage (`outs` field) are not removed by this command,
+unless the `--outs` option is used which will remove **all** of them.
+Alternatively, you can the names of individual <abbr>output</abbr> files or
```

Comment:
but in this case we don't do anything to the .dvc or dvc.yaml? 

this mixed semantics confuses me a bit to be honest

with some targets we touch dvc.yaml, but not data ... with some targets we touch data, but not DVC files 

also, `--outs` that is not clear how it should be have when file/directory path is provided

It feels to me that even if we merge this (and the DVC core one), we'll need to get back to the whiteboard pretty soon with this `dvc remove` that we are long overdue cc @efiop @dberenbaum

Reply:
Overall, not sure if this has enough value to do this now at least - it seems it's pretty much a replacement to `rm -rf` (+git command to remove .gitignore), unless I'm missing something? @daavoo what is your use case that you had in mind?

sorry, that we didn't get back to you initially in the DVC core ticket (or may be I missed the discussion).

Reply:
> Overall, not sure if this has enough value to do this now at least - it seems it's pretty much a replacement to `rm -rf` (+git command to remove .gitignore), unless I'm missing something? @daavoo what is your use case that you had in mind?
> 
> sorry, that we didn't get back to you initially in the DVC core ticket (or may be I missed the discussion).

I agree with the problem of mixed semantics @shcheklein . Sadly I think I'm in no position to explain `why` (an intended use case) but rather `how` (what the new behavior is); I just picked a `Good First Issue` ticket to force myself to dive in the source code :sweat_smile: . 

I think the original idea of use case was https://github.com/iterative/dvc/issues/4497 but I might misinterpreted the use case along the implementation.

Reply:
Yeah, sorry for the lack of clarity here after you've already put in so much work @daavoo.

> but in this case we don't do anything to the .dvc or dvc.yaml?

I just tested, and it looks like .dvc is removed with this command but dvc.yaml won't be touched. Is that right @daavoo?

There might be enough straightforward utility here to merge the PR if it is limited to making `dvc remove data.xml` equivalent to `dvc remove data.xml.dvc` to support https://github.com/iterative/dvc/issues/2575#issuecomment-707868202.

The rest might be valuable, too, but it seems like the semantics need to be straightened out in https://github.com/iterative/dvc/issues/5791 first.

Reply:
If you consider that we need to take a step back and put this and the dvc P.R. on pause or even discarding them in favor of waiting to find the right semantics, I would not mind it at all. 

I wanted to dive into the source code and your thoughtful development process and I did; don't feel obligated for merging something just because of the work i already put.